remoting/zookeeper/listener_test.go

```diff
@@ -86,32 +87,36 @@ func TestListener(t *testing.T) {
 	dubbo.service.com.ikurento.user.UserProvider.warmup=100
 	dubbo.service.com.ikurento.user.UserProvider.cluster=failover
 `
-
+	var wait sync.WaitGroup
 	ts, client, event := initZkData(t)
 	defer ts.Stop()
 	client.Wait.Add(1)
+	wait.Add(1)
 	go client.HandleZkEvent(event)
 	listener := NewZkEventListener(client)
-	dataListener := &mockDataListener{client: client, changedData: changedData}
+	dataListener := &mockDataListener{client: client, changedData: changedData, wait: &wait}
 	listener.ListenServiceEvent("/dubbo", dataListener)
 
 	_, err := client.Conn.Set("/dubbo/dubbo.properties", []byte(changedData), 1)
 	assert.NoError(t, err)
-	client.Wait.Wait()
+	wait.Wait()
 	assert.Equal(t, changedData, dataListener.eventList[1].Content)
+	client.Close()
+
 }
 
 type mockDataListener struct {
 	eventList   []remoting.Event
 	client      *ZookeeperClient
 	changedData string
+	wait        *sync.WaitGroup
 }
 
 func (m *mockDataListener) DataChange(eventType remoting.Event) bool {
 	fmt.Println(eventType)
 	m.eventList = append(m.eventList, eventType)
 	if eventType.Content == m.changedData {
-		m.client.Wait.Done()
```

Comment:
Why not just change `m.client.Wati.Done` to `m.client.Close`?

Reply:
If I want to test another data change event, close client may not be a good idea. Instead , I define a wait group for one data change event

Reply:
@hxmhlt Ok. I think your solution may be more suitable.